src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -90,44 +109,75 @@ protected sealed override void InitializeWorker(AnalysisContext context)
             if (!ShouldAnalyze(context, namedType))
                 return;
 
-            var fieldNames = _fieldNamesPool.Allocate();
+            // Results of our analysis pass that we will use to determine which fields and properties to offer to fixup.
             var analysisResults = s_analysisResultPool.Allocate();
-            var ineligibleFields = s_fieldSetPool.Allocate();
-            var nonConstructorFieldWrites = s_fieldWriteLocationPool.Allocate();
 
-            // Record the names of all the fields in this type.  We can use this to greatly reduce the amount of
+            // Fields whose usage may disqualify them from being removed (depending on the usage location). For example,
+            // a field taken by ref normally can't be converted (as a property can't be taken by ref).  However, this
+            // doesn't apply within the property itself (as it can refer to `field` after the rewrite).
+            var ineligibleFields = s_fieldToUsageLocationPool.Allocate();
+
+            // Locations where this field is read or written.  If it is read or written outside of hte property being
+            // changed, and the property getter/setter is non-trivial, then we cannot use 'field' for it, as that would 
+            // change the semantics in those locations.
+            var fieldReads = s_fieldToUsageLocationPool.Allocate();
+            var fieldWrites = s_fieldToUsageLocationPool.Allocate();
```

Comment:
previously we just tracked writes to fields outside of a constructor.  now we track all reads/writes of hte field anywhere.  Later, we'll slice/dice those results to make the determination if we can't offer the change.